### PR TITLE
Toggle "fullscreen"

### PR DESCRIPTION
**Pull request by cristianiacobanu**
Hello fellow humans,
The pull includes new `paper-toggle-button` dependency, adding simple toggle element visibility.
Hope you like it!

---

Default
---
image

Toggle
---
image

### CODE REVIEW

**Comment by notwaldorf** [comment]
Ugh, I kind of don't want to adfd `typography.html`, and let it use sans-serif if you don't have `Roboto` natively installed. Web fonts are kind of the worst, and I don't really want to add the FOUT just for a slightly fancier font

File: bower.json [hunk 1]
```diff
@@ -17,7 +17,9 @@
     "iron-iconset-svg": "PolymerElements/iron-iconset-svg#^2.0.0",
     "iron-selector": "PolymerElements/iron-selector#^2.0.0",
     "iron-pages": "PolymerElements/iron-pages#^2.0.0",
-    "iron-ajax": "PolymerElements/iron-ajax#^2.0.0"
+    "iron-ajax": "PolymerElements/iron-ajax#^2.0.0",
+    "paper-toggle-button": "PolymerElements/paper-toggle-button#^2.0.0",
+    "paper-styles": "PolymerElements/paper-styles#^2.0.0"
```
Comment (notwaldorf) [inline bower.json]: I would remove that (see comment)

File: elements/app-shell.html [hunk 2]
```diff
@@ -46,6 +48,7 @@
         --highlight-pink: #e91e63;
         --highlight-blue: #2196f3;
         --input-border-color: #596c7a;
+        --header-height-fullscreen: 44px;
```
Comment (notwaldorf) [inline elements/app-shell.html]: The header resizing is actually really jarring imo. Could we leave it the same size, please?

File: elements/app-shell.html [hunk 3]
```diff
@@ -72,6 +75,13 @@
       app-toolbar {
         display: flex;
         justify-content: space-between;
+        transition: all .1s ease-in;
+      }
+
+      app-header[fullscreen],
+      app-header[fullscreen] app-toolbar {
+        font-size: 1em;
+        height: var(--header-height-fullscreen);
```
Comment (notwaldorf) [inline elements/app-shell.html]: Remove this (see comment above)

File: elements/element-stuff-styles.html [hunk 4]
```diff
@@ -27,8 +27,14 @@
...
         line-height: 1.5;
         color: white;
-        background: #232733;
+        background: var(--dark-grey);
         margin-top: 4px;
+        cursor: pointer;
+        outline: none;
+        border-radius: 5px;
+      }
+      button.add:hover {
```
Comment (notwaldorf) [inline elements/element-stuff-styles.html]: cool!

**Comment by cristianiacobanu** [comment]
Roger that. Will prepare it till (US) tomorrow

**Comment by cristianiacobanu** [comment]
@notwaldorf - Removed `typography.html` and header style changes on fullscreen toggle. Toggle button now just sets `.drawer`s to `display: none`.

**Comment by cristianiacobanu** [comment]
@notwaldorf - Added a CSS way of handling focus ring, adding tab focus I previously disabled.
Enabled `overflow-y: auto;` on `tree-view` to allow scrolling when the view is overpopulated.
Formatted `label`, `input` and `buttons` for a bit of a11y :fist_raised:

File: elements/app-controls.html [hunk 5]
```diff
@@ -38,16 +38,16 @@
         margin: 8px;
       }
     </style>
-    <button on-click="undo" id="undoBtn" disabled title="Undo">
+    <button role="button" on-click="undo" id="undoBtn" disabled title="Undo">
```
Comment (notwaldorf) [inline elements/app-controls.html]: tbh i don't think is needed (button has a default role of `button`)

**Comment by notwaldorf** [comment]
I would remove all the `role=button` from the buttons, since they're duplicate, but I'll just merge it since it's been open for far too long :/
(sorry!)

**Pull request merged**